src/server/routes/unitReadings.js

```diff
@@ -466,22 +477,24 @@ function createRouter() {
 		if (!(validateMeterThreeDReadingsParams(req.params) && validateThreeDQueryParams(req.query))) {
 			res.sendStatus(400);
 		} else {
-			// Get time range to validate 1 year or less.
+			// Security: This is an open route (login not needed). Enforce MAX_3D_DAYS limit to prevent abuse.
 			const timeInterval = TimeInterval.fromString(req.query.timeInterval);
 			if (!timeInterval.getIsBounded()) {
 				// Cannot do if not bounded.
 				res.sendStatus(400);
 			} else {
-				const duration = moment.duration(timeInterval.endTimestamp.diff(timeInterval.startTimestamp));
-				// Gets 0 unless one day beyond a year but that okay since don't do partial days.
-				const durationInYears = duration.years();
-				if (durationInYears >= 1) {
-					// Limit 3D to one year of data.
+				// Calculate calendar days (inclusive of both start and end day) to match client calculation
+				// Client uses: threeDEndDate.diff(threeDStartDate, 'days') + 1
+				const durationInDays = timeInterval.endTimestamp.diff(timeInterval.startTimestamp, 'days') + 1;
+				// Security: Limit 3D to MAX_3D_DAYS (1095 days = 3 years) to prevent abuse from unauthenticated users.
+				// Frontend auto-adjustment provides UX, but backend must enforce security limits.
+				if (durationInDays > 1095) {
```

Comment:
As was done on the client, it is nice to make limits a const though it will be a little different due to JS not TS. I think that should be done here, esp. since the value is used more than once.